apps/bam-otf-server/app/queues/transaction.server.ts

```diff
@@ -19,115 +18,116 @@ const QUEUE_ID = 'transaction'
  * This queue checks if the wallet has a new transaction and updates the database accordingly.
  * It also triggers a webhook if the payment was successful.
  */
-export const queue = createQueue<QueueData>(QUEUE_ID, async job => {
-  const {data: payload} = job
-  const {paymentIntentId} = payload
-
-  const log = QueueLog(QUEUE_ID, paymentIntentId)
-  log('started')
-
-  const [transactions, paymentIntent] = await Promise.all([
-    // Check if the payment was made
-    listUnspent(paymentIntentId),
-    prisma.paymentIntent.findFirst({
-      where: {id: paymentIntentId},
-    }),
-  ])
-
-  if (!paymentIntent) {
-    throw new UnrecoverableError(
-      `Payment intent not found: ${format.red(paymentIntentId)}`,
+export const queue = createQueue<QueueData>(
+  QUEUE_ID,
+  async job => {
+    const {data: payload} = job
+    const {paymentIntentId} = payload
+
+    const [transactions, paymentIntent] = await Promise.all([
+      // Check if the payment was made
+      listUnspent(paymentIntentId),
+      prisma.paymentIntent.findFirst({
+        where: {id: paymentIntentId},
+      }),
+    ])
+
+    if (!paymentIntent) {
+      throw new UnrecoverableError(
+        `Payment intent not found: ${format.red(paymentIntentId)}`,
+      )
+    }
+
+    if (paymentIntent.status === 'canceled') {
+      throw new UnrecoverableError(
+        `Payment intent was canceled ${format.red(paymentIntentId)}`,
+      )
```

Comment:
The main change is the addition of an additional payment status check. After querying the database, the code checks whether the status field of the paymentIntent object is equal to "canceled". If so, the code throws an UnrecoverableError exception stating that the payment was cancelled.